configure

```diff
@@ -0,0 +1,338 @@
+#!/bin/sh
+### Custom configure script of Dinit.
+## Initial prepartion
+set -eu
+cd "$(dirname "$0")"
+. build/version.conf
+
+## Helper funcations
+# See http://www.etalabs.net/sh_tricks.html
+echo()
+{
+    IFS=" " printf %s\\n "$*"
+}
+
+info()
+{
+    echo "Info: $*"
+}
+
+sub_info()
+{
+    echo "  ... $*"
+}
+
+error()
+{
+    echo "Error: $*"
+    exit 1
+}
+
+sub_error()
+{
+    echo "  ... Error: $*"
+    exit 1
+}
+
+warning()
+{
+    echo
+    echo "Warning: $*"
+}
+
+sub_warning()
+{
+    echo "  ... $*"
+    echo
+}
+
+create_testfile_cc()
+{
+    echo "int main(int argc, char **argv) { return 0; }" > testfile.cc
+}
+
+findcxx()
+{
+    if type "$1" > /dev/null 2>&1; then
+        if [ "${2:-}" = "host" ]; then
+            hostcompiler="$1"
+        else
+            compiler="$1"
+        fi
+    fi
+}
+
+cxx_works()
+{
+    info Checking "$1" works...
+    if ! $compiler -o testfile.o testfile.cc; then
+        rm -f testfile*
+        sub_error Seems like C++ compiler is not working!
+    else
+        rm -f testfile
+        sub_info Yes.
+    fi
+}
+
+find_program()
+{
+    if [ -x $(command -v "$1") ]; then
+        sub_info "$(command -v "$1")"
+    else
+        sub_error "$1" is not executable/exist!
+    fi
+}
+
+try_cxx_argument()
+{
+    info Checking whatever compiler accpets "$2"...
+    if $compiler -c $CXXOPTS $2 ${POST_CXXOPTS:-} testfile.cc -o testfile.o > /dev/null 2>&1; then
```

Comment:
The makefiles currently use CXXOPTS for compilation _and_ linking. So you need to make sure it works for the options work for both.